### PR TITLE
Add LaTeX complex numbers representation

### PR DESCRIPTION
**Pull request by aleksandr-bakanov**
Complex numbers are presented by Quantity::_repr_latex_() as (real +- imag i), where real and imag parts are represented by Latex.format_exponential_notation.

### CODE REVIEW

**Comment by bsipocz** [comment]
@aleksandr-bakanov - The test failure is real, there seems to be some numpy version dependent behaviour difference.

Also, could you please add a changelog entry?

File: astropy/units/quantity.py [hunk 1]
```diff
@@ -1088,8 +1088,16 @@ def float_formatter(value):
             return Latex.format_exponential_notation(value,
                                                      format_spec=format_spec)
 
+        def complex_formatter(value):
+            return '({0}{1}i)'.format(
+                Latex.format_exponential_notation(value.real,
+                                                     format_spec=format_spec),
```
Comment (mhvk) [inline astropy/units/quantity.py]: Sorry to be annoying, but could you align the indentation... (here and in line below)

Comment (aleksandr-bakanov) [reply astropy/units/quantity.py]: Sure, no problem

**Comment by aleksandr-bakanov** [comment]
It is worth to notice that this feature is related to #5071 . Though I don't know yet how to format record arrays (and even what is a record). Primarily I've thought about implementation of `Quantity.to_string` but stopped on current solution for complex numbers.

**Comment by aleksandr-bakanov** [comment]
@mhvk thanks for clarification about failed tests! Rebase is done.

**Pull request merged**